cluster.go

```diff
@@ -143,82 +154,65 @@ func (c *ClusterClient) process(cmd Cmder) {
 	}
 }
 
-// Closes all connections and reloads slot cache, if due
-func (c *ClusterClient) reloadIfDue() (err error) {
-	if !atomic.CompareAndSwapUint32(&c._reload, 1, 0) {
-		return
+// Closes all clients and returns last error if there are any.
+func (c *ClusterClient) resetClients() (err error) {
+	c.clientsMx.Lock()
+	for addr, client := range c.clients {
+		if e := client.Close(); e != nil {
+			err = e
+		}
+		delete(c.clients, addr)
 	}
+	c.clientsMx.Unlock()
+	return err
+}
 
-	var infos []ClusterSlotInfo
-
+func (c *ClusterClient) setSlots(slots []ClusterSlotInfo) {
 	c.slotsMx.Lock()
-	defer c.slotsMx.Unlock()
 
-	// Try known addresses in random order (map interation order is random in Go)
-	// http://redis.io/topics/cluster-spec#clients-first-connection-and-handling-of-redirections
-	// https://github.com/antirez/redis-rb-cluster/blob/fd931ed/cluster.rb#L157
-	for addr := range c.addrs {
-		c.reset()
+	c.addrs = c.addrs[:0]
```

Comment:
I remember this being discouraged by the cluster spec. You should always remember all known addresses. The edge case is `slots` being empty because of a broken or detached node. Then, you would have nothing to fall back on.